app/CMakeLists.txt

```diff
@@ -143,19 +143,26 @@ endif()
 if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_CUDA AND CELERITAS_USE_VecGeom)
   # Since the demo kernel links against VecGeom, which requires CUDA separable
   # compilation, it cannot be linked directly into an executable.
-  add_library(celeritas_demo_rasterizer
+  celeritas_add_library(celeritas_demo_rasterizer
     demo-rasterizer/RDemoRunner.cc
     demo-rasterizer/RDemoKernel.cu
     demo-rasterizer/ImageIO.cc
     demo-rasterizer/ImageStore.cc
+    SHARED
   )
   target_link_libraries(celeritas_demo_rasterizer
     PRIVATE
-      celeritas
+      Celeritas::Core
       nlohmann_json::nlohmann_json
   )
+
   celeritas_link_vecgeom_cuda(celeritas_demo_rasterizer)
 
+  # Note: the repeat (target name, location) below is due the author of  cuda_add_library_depend
+  # not knowing how to automatically go from the target to the real file from a generator expression in add_custom_command
+  get_property(vecgeom_static_target_location TARGET VecGeom::vecgeomcuda_static PROPERTY LOCATION)
+  cuda_add_library_depend(celeritas_demo_rasterizer celeritas_static ../src/libceleritas_static.a VecGeom::vecgeomcuda_static ${vecgeom_static_target_location})
```

Comment:
Identical code shows up in a few places; can this not be rolled into `celeritas_link_vecgeom_cuda`? If not, can you make a separate helper function/macro?